tango/workspaces/local_workspace.py

```diff
@@ -362,6 +362,38 @@ def registered_runs(self) -> Dict[str, Run]:
             if run_dir.is_dir()
         }
 
+    def search_step_info(
+        self,
+        *,
+        sort_by: StepInfoSort = StepInfoSort.CREATED,
+        sort_descending: bool = True,
+        match: Optional[str] = None,
+        state: Optional[StepState] = None,
+        start: int = 0,
+        stop: Optional[int] = None,
+    ) -> List[StepInfo]:
+        with SqliteDict(self.step_info_file, flag="r") as d:
+            steps = [
+                step
+                for step in d.values()
+                if (match is None or match in step.unique_id)
+                and (state is None or step.state == state)
+            ]
```

Comment:
I'm not a fan of how this has to read every step every time, but I guess there is no way around that. We'd have to change how these are stored.